front/types/chat.ts

```diff
@@ -12,18 +12,26 @@ export type ChatRetrievedDocumentType = {
   }[];
 };
 
+export type MessageFeedbackStatus = "positive" | "negative" | null;
+
+export type MessageRole = "user" | "retrieval" | "assistant" | "error";
 export type ChatMessageType = {
-  role: "user" | "retrieval" | "assistant" | "error";
+  sId: string;
+  role: MessageRole;
   message?: string; // for `user`, `assistant` and `error` messages
   retrievals?: ChatRetrievedDocumentType[]; // for `retrieval` messages
   query?: string; // for `retrieval` messages (not persisted)
+  feedback?: MessageFeedbackStatus;
 };
 
-export type ChatSessionType = {
+export type LightChatSessionType = {
   id: number;
   userId: number;
   created: number;
   sId: string;
   title?: string;
```

Comment:
Let's just make `messages` optional with a `?` instead of having 2 types?

Agree it provides more safety but makes the code harder to understand / is more complex